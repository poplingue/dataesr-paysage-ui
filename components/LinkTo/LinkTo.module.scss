.LinkTo {
    a {
        color: var(--blue-france-sun-113-625);
        box-shadow: none;
        padding-bottom: 3px;
    }
}
