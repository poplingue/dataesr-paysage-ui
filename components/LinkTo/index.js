import { Icon } from '@dataesr/react-dsfr';
import PropTypes from 'prop-types';
import NavLink from '../NavLink';
import styles from './LinkTo.module.scss';

export default function LinkTo({ href, text, icon }) {
    return (
        <div className={styles.LinkTo}>
            <NavLink href={href}>
                <Icon iconPosition="right" name={icon} size="lg">
                    <>{text}</>
                </Icon>
            </NavLink>
        </div>
    );
}

LinkTo.defaultProps = {
    icon: 'ri-arrow-right-line',
};

LinkTo.propTypes = {
    href: PropTypes.string.isRequired,
    icon: PropTypes.string,
    text: PropTypes.string.isRequired,
};
