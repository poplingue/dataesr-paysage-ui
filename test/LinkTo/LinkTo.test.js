import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import LinkTo from '../../components/LinkTo';

Enzyme.configure({ adapter: new Adapter() });

describe('<LinkTo />', () => {
    let wrapper;
    const props = {
        text: 'Ceci est mon texte',
        href: '/this/is/my/href',
    };

    beforeEach(() => {
        wrapper = (props = {}) =>
            shallow(<LinkTo text={props.text} href={props.href} />);
    });
    it('renders correctly LinkTo', () => {
        const tree = renderer
            .create(
                <LinkTo text="Ceci est mon texte" href="/contrib/structure" />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has LinkTo class', () => {
        const component = wrapper(props);
        expect(component.find('div').hasClass('LinkTo')).toBeTruthy();
    });
});
