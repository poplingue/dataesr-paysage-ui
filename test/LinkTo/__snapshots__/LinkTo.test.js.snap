// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkTo /> renders correctly LinkTo 1`] = `
<div
  className="LinkTo"
>
  <a
    href="/update/structure"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <span>
      Ceci est mon texte
      <span
        aria-hidden={true}
        className="ri-lg icon-right ds-fr--v-middle ri-arrow-right-line"
      />
    </span>
  </a>
</div>
`;
